Orange/base.py

```diff
@@ -320,12 +320,55 @@ def backmap_probs(self, probs, n_values, backmappers):
         new_probs = new_probs / tots[:, None]
         return new_probs
 
+    def data_to_model_domain(self, data: Table) -> Table:
+        """
+        Transforms data to the model domain if possible.
+
+        Parameters
+        ----------
+        data
+            Data to be transformed to the model domain
+
+        Returns
+        -------
+        Transformed data table
+
+        Raises
+        ------
+        DomainTransformationError
+            Error indicates that transformation is not possible since domains
+            are not compatible
+        """
+        if data.domain == self.domain:
+            return data
+
+        if self.original_domain.attributes != data.domain.attributes \
+                and data.X.size \
+                and not all_nan(data.X):
+            new_data = data.transform(self.original_domain)
+            if all_nan(new_data.X):
+                raise DomainTransformationError(
+                    "domain transformation produced no defined values")
+            return new_data.transform(self.domain)
+
+        return data.transform(self.domain)
+
     def __call__(self, data, ret=Value):
         multitarget = len(self.domain.class_vars) > 1
 
         def one_hot_probs(value):
             if not multitarget:
-                return one_hot(value)
+                probs = one_hot(value)
+                # sometimes a maximum class value(s) are not predicted and
+                # those probs.shape[1] does not match to the length of values
+                # in the domain - here we add missing dimension if required
+                num_val = (
+                    len(self.original_domain.class_var.values)
+                    if self.original_domain is not None else probs.shape[0]
+                )
+                return np.hstack((
+                    probs, np.zeros((probs.shape[0], num_val - probs.shape[1]))
+                ))
```

Comment:
The only other use of `one_hot` I found, had an almost identical (but not the same) fix as here.
It would probably be better to improve `one_hot` itself instead of calling it and adding a fix after that every time.
E.g. an additional optional argument like `max_value` which is used if provided (+ maybe a check that if `max_value` is given it must be `>=` the max found in values).